task_definition.tf

```
@@ -0,0 +1,4 @@
+resource "aws_ecs_task_definition" "task_definition" {
+  family                = "worker"
+  container_definitions = data.template_file.task_definition_template.rendered
```

```suggestion
  container_definitions = templatefile("${path.module}/task_definition.json.tpl",
    {
      REPOSITORY_URL = replace(aws_ecr_repository.worker.repository_url, "https://", "")
    }
  )
```
I dont think you need to use the template_file data block as there is also the templatefile function builtin